libtenzir/builtins/operators/top_rare.cpp

```diff
@@ -79,12 +82,37 @@ class top_rare_plugin final : public virtual operator_parser_plugin {
     return std::move(*parsed);
   }
 
+  auto
+  make(invocation inv, session ctx) const -> failure_or<operator_ptr> override {
+    auto expr = ast::expression{};
+    auto count_field = ast::expression{};
+    TRY(argument_parser2::operator_(name())
+          .add(expr, "<expr>")
+          .add(count_field, "<expr>")
+          .parse(inv, ctx));
+    // TODO: Replace this textual parsing with a subpipeline to improve
+    // diagnostics for this operator.
+
+    // XXX: How to `ast::expression` -> `std::string`?
+    auto repr = fmt::format("summarize {0}=count(.) by {1} | sort {3}{0}",
+                            count_field, expr, Mode == mode::top ? "-" : "");
```

Comment:
This has a problem: It calls TQL1's `summarize`, which has a different notion of what a "field" is. You likely need to create the actual AST here for compile it to a pipeline for TQL2.